carsus/io/cmfgen/base.py

```diff
@@ -809,3 +839,153 @@ def _get_levels_lines(self, data):
+            will have all the temperatures from the CMFGEN dataset by default.
+        """
+        col_list, t_grid = [], []
+        levels_combine = self.levels.copy().reset_index()
+        label_ind_mapping = {
+            label: index for label, index in zip(levels_combine.label, levels_combine.level_index)
+        }
+
+        for ion, data_dict in data.items():
+            levels = data_dict["levels"].copy()
+            collisions = data_dict["collisions"].copy()
+            
+            label_g_mapping = {
+                label: g for label, g in zip(levels.label, levels.g)
+            }
+            missing_labels = set()
+
+            gi, lower_level_index, upper_level_index = [], [], []
+
+            for ll, ul in zip(collisions.label_lower, collisions.label_upper):  
+                if ll in label_ind_mapping:
+                    lower_level_index.append(label_ind_mapping[ll])
+                else:
+                    if not drop_mismatched_labels:
+                        raise KeyError(
+                            f"Label {ll} for ion {ion} could not be mapped. "
+                            "Please check the atomic data files."
+                        )
+                    missing_labels.add(ll)
+                    lower_level_index.append(np.nan)
+                
+                if ll in label_g_mapping:
+                    gi.append(label_g_mapping[ll])
+                else:
+                    if not drop_mismatched_labels:
+                        raise KeyError(
+                            f"Label {ll} for ion {ion} could not be mapped. "
+                            "Please check the atomic data files."
+                        )
+                    missing_labels.add(ll)
+                    gi.append(np.nan)
+
+                if ul in label_ind_mapping:
+                    upper_level_index.append(label_ind_mapping[ul])
+                else:
+                    if ul!='I':
+                        if not drop_mismatched_labels:
+                            raise KeyError(
+                                f"Label {ll} for ion {ion} could not be mapped. "
+                                "Please check the atomic data files."
+                            )
+                        missing_labels.add(ul)
+                    else:
+                        logger.info("Dropping collisional ionization data.")
+                    
+                    upper_level_index.append(np.nan)
+
+            if missing_labels:
+                logger.info(f"Entries having label(s): {', '.join(missing_labels)} will be dropped for ion: {ion}.")
+            
+            collisions["level_number_lower"] = lower_level_index
+            collisions["level_number_upper"] = upper_level_index
+            collisions["gi"] = gi
+            
+            if drop_mismatched_labels:
+                collisions = collisions.dropna(subset=['level_number_lower', 'level_number_upper'])
```

Comment:
```suggestion
            collisions = collisions.dropna(subset=['level_number_lower', 'level_number_upper'])
```

Reply:
Currently, the code crashes for `drop_mismatched_labels=False` because `I` is not dropped. (The code tries to convert a nan to an int in line 926). I don't think there should be any other nan in the dataframe for `drop_mismatched_labels=False` so we can probably just remove the if statement?